repository/src/main/java/org/apache/atlas/repository/store/graph/v2/preprocessor/datamesh/DataDomainPreProcessor.java

```diff
@@ -412,6 +409,12 @@ private void validateStakeholderRelationship(AtlasEntity entity) throws AtlasBas
         }
     }
 
+    private void isParentDomainMovedToChild(String targetDomainQualifiedName, String currentDomainQualifiedName) throws AtlasBaseException {
+        if(targetDomainQualifiedName.contains(currentDomainQualifiedName)){
```

Comment:
Use `.startsWith` instead of contains

Assumptions:
targetDomainQualifiedName - Child domain QN
currentDomainQualifiedName - Super domain QN which is being moved